packages/apps/desktop-viewer-test/package.json

```diff
@@ -84,8 +84,8 @@
     "@types/electron-devtools-installer": "^2.2.0",
     "@types/minimist": "^1.2.0",
     "@types/node": "^18.12.0",
-    "@types/react": "^17.0.19",
-    "@types/react-dom": "^17.0.9",
+    "@types/react": "^18.2.9",

```

Comment:
Have you tested the changes with react 17 as well?

Reply:
Do you mean just change the versions in the package.json back to 17 and try out the changes?

Reply:
Yep

Reply:
Will try that out now

Reply:
One of the changes in the upgrade to react 18 uses `CreateRoot` for rendering components. This is newly available in react 18, so this is not compatible with react 17

Reply:
This is either a major change then, or we detect which version of react is being used and either use CreateRoot or the old method. I believe you can use React.version to detect. 
@aruniverse thoughts?

Reply:
This is entering hacking hell, (hence the `UiFramework.childWindows.useCreateRoot` method), because you need to import from a different package depending on the version, which is something Webpack do not support...

Reply:
Ahh yeah.

Reply:
this is fine since this is in the test app

Reply:
Aren't our peer deps a lie then?

Reply:
i dont think so; our peers say we the viewer pkgs are compatible with react 17 and now additionaly/or react 18
the only actual use of react 18 apis in this pr is the react-dom/client.createRoot done in the test apps